openeo/rest/job.py

```diff
@@ -264,6 +265,13 @@ def __init__(self, job: RESTJob):
     def __repr__(self):
         return "<JobResults for job {j!r}>".format(j=self._job.job_id)
 
+    def _repr_html_(self):
+        try:
+            response = self.get_metadata()
+            return render_component("batch-job-result", data = response)
+        except OpenEoApiError as error:
+            return render_error(error, "Job not finished yet")
```

Comment:
There could also be other errors, I would only catch the `JobNotFinished` ones, e.g.:

```python
try:
    response = self.get_metadata()
except OpenEoApiError as error:
    if error.code == "JobNotFinished":
        return render_error(error, "Job not finished yet")
    raise
else:
    return render_component("batch-job-result", data = response)


(the empty `raise` reraises the existing exception)

Reply:
That is handled, usually, it should just show the message. The message I'm passing there is just a fallback (maybe we don't even need the fallback?)

Reply:
Yeah, I think I'll just remove the fallback.

Reply:
my point is that you should, I think, only catch the "JobNotFinished" error, and let other issues propagate normally

Reply:
Why not show the others nicely? It's just errors that really come from the API, usually with a message, right? I'm not catching all errors, I think.

Reply:
ok, true. It's a bit a gray zone: the user does not call `_repr_html_` knowingly, so maybe it's fine to intercept exceptions a bit more greedily.